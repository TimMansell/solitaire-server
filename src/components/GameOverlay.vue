<template>
  <div class="game-overlay" :class="classes" data-test="game-overlay">
    <div class="game-overlay__container">
      <div class="game-overlay__content">
        <Logo class="game-overlay__logo" v-if="showLogo" />
        <h1 class="game-overlay__title">
          <slot name="title" />
        </h1>
        <div class="game-overlay__msg" v-if="hasMsgSlot">
          <slot name="msg" />
        </div>
        <div class="game-overlay__btns" data-test="game-overlay-btns">
          <slot name="buttons" />
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Logo from './Logo.vue';

export default {
  name: 'GameOverlay',
  components: {
    Logo,
  },
  props: {
    alt: {
      type: Boolean,
      default: false,
    },
    centerContent: {
      type: Boolean,
      default: false,
    },
    showLogo: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    classes() {
      const { alt, centerContent } = this;

      return {
        'game-overlay--alt': alt,
        'game-overlay--centered': centerContent,
      };
    },
    hasMsgSlot() {
      return !!this.$slots.msg;
    },
  },
};
</script>

<style lang="scss" scoped>
.game-overlay {
  display: grid;
  align-items: center;
  position: fixed;
  left: 0;
  top: 0;
  width: 100vw;
  height: 100vh;
  background: rgba($col-tertiary, 0.7);
  z-index: var(--z-overlay);
  overflow-y: auto;

  &--alt {
    background: var(--bg-primary) url('~@/assets/felt.png') repeat;
  }

  &--centered {
    text-align: center;
  }

  &__container {
    justify-self: center;

    @media (min-width: $bp-md) {
      width: 80%;
    }

    @media (min-width: $bp-xl) {
      width: 40%;
    }
  }

  &__logo {
    @media (orientation: landscape) and (pointer: coarse) {
      display: none;
    }

    @media (orientation: landscape) and (min-width: $bp-lg) {
      display: inherit;
    }
  }

  &__content {
    padding: var(--pd-lg);
  }

  &__title {
    color: var(--text-primary);
    text-shadow: -1px -1px rgba($col-tertiary, 0.3);
  }

  &__msg {
    color: var(--text-primary);
    text-shadow: -1px -1px rgba($col-tertiary, 0.3);
    margin-bottom: var(--mg-lg);
  }

  &__btns {
    display: flex;
    justify-content: center;
    gap: 1rem;
  }
}
</style>
