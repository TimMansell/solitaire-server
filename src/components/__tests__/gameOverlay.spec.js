import { shallowMount } from '@vue/test-utils';
import GameOverlay from '@/components/GameOverlay.vue';

const mocks = {
  $store: { dispatch: jest.fn() },
};

describe('GameOverlay.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(GameOverlay, {
      mocks,
      propsData: {
        showLogo: true,
        btnClose: () => {},
      },
      slots: {
        title: 'Title',
        msg: 'Lorem ipsum dolor, sit amet consectetur adipisicing elit.',
        buttons: 'Buttons',
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('does not render logo', () => {
    const wrapper = shallowMount(GameOverlay, {
      mocks,
    });

    expect(wrapper.find('[data-test="game-overlay-logo"]').exists()).toBe(
      false
    );
  });

  it('does not render a close button', () => {
    const wrapper = shallowMount(GameOverlay, {
      mocks,
    });

    expect(wrapper.find('[data-test="game-overlay-close"]').exists()).toBe(
      false
    );
  });

  it('renders an alternate overlay class', () => {
    const wrapper = shallowMount(GameOverlay, {
      mocks,
      propsData: {
        alt: true,
      },
    });

    expect(wrapper.classes()).toContain('game-overlay--alt');
  });

  it('renders an center content class', () => {
    const wrapper = shallowMount(GameOverlay, {
      mocks,
      propsData: {
        centerContent: true,
      },
    });

    expect(wrapper.classes()).toContain('game-overlay--centered');
  });
});
