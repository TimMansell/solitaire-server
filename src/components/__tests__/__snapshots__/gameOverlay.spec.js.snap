// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOverlay.vue does not render logo 1`] = `
<div data-test="game-overlay" class="game-overlay">
  <div class="game-overlay__container">
    <div class="game-overlay__content">
      <!---->
      <h1 class="game-overlay__title"></h1>
      <!---->
      <div data-test="game-overlay-btns" class="game-overlay__btns"></div>
    </div>
  </div>
</div>
`;

exports[`GameOverlay.vue matches snapshot 1`] = `
<div data-test="game-overlay" class="game-overlay">
  <div class="game-overlay__container">
    <div class="game-overlay__content">
      <!---->
      <h1 class="game-overlay__title"></h1>
      <!---->
      <div data-test="game-overlay-btns" class="game-overlay__btns"></div>
    </div>
  </div>
</div>
`;

exports[`GameOverlay.vue renders text in slots 1`] = `
<div data-test="game-overlay" class="game-overlay">
  <div class="game-overlay__container">
    <div class="game-overlay__content">
      <!---->
      <h1 class="game-overlay__title">title</h1>
      <div class="game-overlay__msg">msg</div>
      <div data-test="game-overlay-btns" class="game-overlay__btns">buttons</div>
    </div>
  </div>
</div>
`;
